src/decisionengine/framework/modules/tests/test_de_logger.py

```diff
@@ -1,49 +1,64 @@
-
-    def test_by_nonsense_is_err(self):
-        with pytest.raises(ValueError) as err:
-            with tempfile.NamedTemporaryFile() as log:
-                de_logger.set_logging(log_level='INFO', max_backup_count=6,
-                                      file_rotate_by="nonsense", max_file_size=1000000,
-                                      log_file_name=log.name)
-        assert "Incorrect 'file_rotate_by'" in str(err.value)
-
-    def test_by_size(self):
-        with tempfile.NamedTemporaryFile() as log:
-            mylogger = de_logger.set_logging(log_level='INFO', max_backup_count=6,
-                                             file_rotate_by="size", max_file_size=1000000,
-                                             log_file_name=log.name)
 
-            assert mylogger.hasHandlers() is True
-            assert 'RotatingFileHandler' in str(mylogger.handlers)
-            de_logger.set_stream_logging()
-            assert mylogger.debug('debug') is None
-            assert mylogger.info('infomsg') is None
+@pytest.fixture
+def log_setup(request):
+    my_log = structlog.getLogger("decision_engine")
+
+    # make sure it is in a known "unconfigured state"
+    while len(my_log.handlers) > 0:
+        my_log.removeHandler(my_log.handlers[0])
+
+    yield my_log
+
+    # make sure we leave this without any handlers
+    while len(my_log.handlers) > 0:
+        my_log.removeHandler(my_log.handlers[0])
+
+    gc.collect()
+
 
-    def test_by_time(self):
+@pytest.mark.usefixtures("log_setup")
+def test_by_nonsense_is_err(log_setup):
+    with pytest.raises(ValueError) as err:
         with tempfile.NamedTemporaryFile() as log:
-            mylogger = de_logger.set_logging(log_level='INFO', rotation_interval=1,
-                                             file_rotate_by="time", rotation_time_unit='D',
-                                             log_file_name=log.name)
-
-            assert mylogger.hasHandlers() is True
-            assert 'TimedRotatingFileHandler' in str(mylogger.handlers)
-            de_logger.set_stream_logging()
-            assert mylogger.debug('debug') is None
-            assert mylogger.info('infomsg') is None
+            de_logger.set_logging(
+                log_level="INFO",
+                max_backup_count=6,
+                file_rotate_by="nonsense",
+                max_file_size=1000000,
+                log_file_name=log.name,
+            )
+    assert "Incorrect 'file_rotate_by'" in str(err.value)
+
+
+@pytest.mark.usefixtures("log_setup")
+def test_by_size(log_setup):
+    with tempfile.NamedTemporaryFile() as log:
+        de_logger.set_logging(
+            log_level="INFO", max_backup_count=6, file_rotate_by="size", max_file_size=1000000, log_file_name=log.name
+        )
+
+        assert log_setup.hasHandlers() is True
+        assert "RotatingFileHandler" in str(log_setup.handlers)
+        assert log_setup.debug("debug") is None
+        assert log_setup.info("infomsg") is None
```

Comment:
I don't understand the goal of these 2 messages. The return code of and logging is always None. Is this to check for exceptions in the eecution?
To test the logs content structlog provides structlog.testing.LogCapture.
Same comment applies for test_by_time below

Reply:
We have agreed to fix the testing in the future. For now we are skipping these tests.